tests/vlan/test_vlan.py

```diff
@@ -35,71 +33,141 @@ def cfg_facts(duthosts, rand_one_dut_hostname):
+def vlan_intfs_dict(cfg_facts, tbinfo):
+    vlan_intfs_dict = {}
+    if tbinfo['topo']['name'] == 't0-56-po2vlan':
+        for k, v in cfg_facts['VLAN'].items():
+            vlanid = v['vlanid']
+            for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+                if addr.find(':') == -1:
+                    ip = addr
+                    break
+            else:
+                continue
+            logger.info("Original VLAN {}, ip {}".format(vlanid, ip))
+            vlan_intfs_dict[int(vlanid)] = {'ip': ip, 'orig': True}
+    if tbinfo['topo']['name'] != 't0-56-po2vlan':
+        vlan_cnt = 0
+        for i in xrange(0, 255):
+            vid = 100 + i
+            if vid in vlan_intfs_dict:
+                continue
+            ip = u'192.168.{}.1/24'.format(i)
+            for v in vlan_intfs_dict.values():
+                if compare_network(ip, v['ip']):
+                    break
+            else:
+                logger.info("Add VLAN {}, ip {}".format(vid, ip))
+                vlan_intfs_dict[vid] = {'ip': ip, 'orig': False}
+                vlan_cnt += 1
+            if vlan_cnt >= 2:
+                break
+    return vlan_intfs_dict
 
 
 @pytest.fixture(scope="module")
-def vlan_ports_list(rand_selected_dut, tbinfo, cfg_facts):
+def vlan_ports_list(duthosts, rand_one_dut_hostname, rand_selected_dut, tbinfo, cfg_facts, vlan_intfs_dict):
+    duthost = duthosts[rand_one_dut_hostname]
     mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
     vlan_ports_list = []
     config_ports = {k: v for k,v in cfg_facts['PORT'].items() if v.get('admin_status', 'down') == 'up'}
     config_portchannels = cfg_facts.get('PORTCHANNEL', {})
     config_port_indices = {k: v for k, v in mg_facts['minigraph_ptf_indices'].items() if k in config_ports}
     ptf_ports_available_in_topo = {port_index: 'eth{}'.format(port_index) for port_index in config_port_indices.values()}
-    config_port_channel_members = [port_channel[1]['members'] for port_channel in config_portchannels.items()]
+    config_port_channel_members = [port_channel['members'] for port_channel in config_portchannels.values()]
     config_port_channel_member_ports = list(itertools.chain.from_iterable(config_port_channel_members))
+    config_ports_vlan = defaultdict(list)
+    vlan_members = cfg_facts.get('VLAN_MEMBER', {})
+    for k, v in cfg_facts['VLAN'].items():
+        vlanid = v['vlanid']
+        for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+            if addr.find(':') == -1:
+                ip = addr
+                break
+        else:
+            continue
+        for port in v['members']:
+            if k in vlan_members and port in vlan_members[k]:
+                mode = vlan_members[k][port]['tagging_mode']
+                config_ports_vlan[port].append((int(vlanid), ip, mode))
+
+    vlan_id_list = [k for k, v in vlan_intfs_dict.items() if v['orig'] == False]
```

Comment:
>orig [](http://example.com/codeflow?start=64&length=4)

I cannot any code relating this field. Add some comment?

Reply:
Refined the question.

Reply:
I see it is introduce in your new code `def vlan_intfs_dict(cfg_facts, tbinfo):`